---
title: "Untitled"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#LOAD LIBRARIES #
```{r}
library(tidyr)
library(tidyverse)
library(dplyr)
library(ggplot2)
```

# IMPORT DATA SHEETS #
```{r}
pre_seq <- readxl::read_xlsx("data/Sequencing/Submission_and_Return_Sheets/2021-09_Voolstra-Barshis_Mote_Samples_withQC.xlsx") %>%
  mutate(origin = "post")

post_seq <- readxl::read_xlsx("data/Sequencing/Submission_and_Return_Sheets/H204SC21103772_DataQC.xlsx") %>%
  mutate(origin = "post")

need_to_add <- readxl::read_xlsx("data/Sequencing/Submission_and_Return_Sheets/Need_to_add_DataQC.xlsx") %>%
  mutate(origin = "need to add")

post_x <- right_join(pre_seq, post_seq, by = "Sample") %>%
  mutate(seq_name = paste("R",Sample, sep = ''))

library("writexl")
write_xlsx(post_x,"data/Sequencing/R260_combined.xlsx")

need_to_add_x <- right_join(pre_seq, need_to_add, by = "Sample") %>%
    mutate(seq_name = paste("R",Sample, sep = ''))

write_xlsx(need_to_add_x,"data/Sequencing/needtoadd_combined.xlsx")


x %>%
  filter(origin == "post")

```


# RAW FASTQ FILE NAMES #
```{r}
# Raw list from cluster after find and replacing "_1.fq.gz"
R_1 <- read.delim("data/Sequencing/Raw_Fastqs/raw_fastqs_sample_names_1.txt")

# Raw list from cluster after find and replacing "_2.fq.gz"
R_2 <- read.delim("data/Sequencing/Raw_Fastqs/raw_fastqs_sample_names_2.txt")

# Sample_ID list taken from datasheet returned from novogene 
sample_IDs <- read.delim("data/Sequencing/Raw_Fastqs/sample_IDs.txt")

setdiff(R_1$fastq, sample_IDs$fastq)

```

# SAMPLE NAME TABLE #
```{r}
# Combine tables 
df <- right_join(R_1, sample_IDs)

# R_1 
R_1_rename_ready <- df %>%
  select(fastq, RNASeq_R1) %>%
  mutate(OldName = paste(fastq, "_1.fq.gz", sep = ""),
         NewName = paste(RNASeq_R1, ".fq.gz", sep = "")) %>%
  select(OldName, NewName)

write.table(R_1_rename_ready, file = "data/Sequencing/Raw_Fastqs/R_1_fastq_rename.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = TRUE)

# R_2
R_2_rename_ready <- df %>%
  select(fastq, RNASeq_R2) %>%
  mutate(OldName = paste(fastq, "_2.fq.gz", sep = ""),
         NewName = paste(RNASeq_R2, ".fq.gz", sep = "")) %>%
  select(OldName, NewName)

write.table(R_2_rename_ready, file = "data/Sequencing/Raw_Fastqs/R_2_fastq_rename.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = TRUE)

```

#### PREVIOUS NAME TABLE CODE #####
# RAW FASTQ FILE NAMES #

```{r}
# Raw list from cluster after find and replacing "_1.fq.gz"
R_1 <- read.delim("data/Sequencing/Raw_Fastqs/raw_fastqs_sample_names_1.txt")

# Raw list from cluster after find and replacing "_2.fq.gz"
R_2 <- read.delim("data/Sequencing/Raw_Fastqs/raw_fastqs_sample_names_2.txt")

# Sample_ID list taken from datasheet returned from novogene 
sample_names <- read.delim("data/Sequencing/Raw_Fastqs/sample_names.txt")

setdiff(R_1$fastq, sample_names$fastq)

```

# R_1 TABLE #
```{r}
# Joining the raw cluster list with the sample IDs for R_1 and adding _1.fq.gz file name
R_1_rename <- right_join(R_1, sample_names, by = "fastq") %>%
  mutate(old_fastq = paste(fastq, "_1.fq.gz", sep = "")) %>%
  mutate(new_fastq = paste(Sample_ID, "_1.fq.gz", sep = ""))

# Final Sample Table for R_1
R_1_rename_ready <- R_1_rename %>%
  select(old_fastq, new_fastq)

# Export Table 
write.table(R_1_rename_ready, file = "data/Sequencing/Raw_Fastqs/R_1_fastq_rename.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

```

# R_2 TABLE #
```{r}
# Joining the raw cluster list with the sample IDs for R_1 and adding _2.fq.gz file name
R_2_rename <- right_join(R_2, sample_names, by = "fastq") %>%
  mutate(old_fastq = paste(fastq, "_2.fq.gz", sep = "")) %>%
  mutate(new_fastq = paste(Sample_ID, "_2.fq.gz", sep = ""))

# Final Sample Table for R_2
R_2_rename_ready <- R_2_rename %>%
  select(old_fastq, new_fastq)

# Export Table 
write.table(R_2_rename_ready, file = "data/Sequencing/Raw_Fastqs/R_2_fastq_rename.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

```

# R_1 SAMPLE NAME SHEET #
```{r}
R_1_file_names <- R_1 %>%
  mutate(fastq = paste(Novogene_R, "_1.fq.gz", sep = "")) %>%
  mutate(new_fastq = paste(Sample_ID, "_1.fq.gz", sep = ""))

R_1_rename_ready <- right_join(R_1, R_1_file_names, by = "fastq") %>%
  select(fastq, new_fastq)

write.table(R_1_rename_ready, file = "data/Sequencing/Raw_Fastqs/R_1_fastq_rename.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

```

# R_2 SAMPLE NAME SHEET #
```{r}
R_2_file_names <- sample_names %>%
  mutate(fastq = paste(Novogene_R, "_2.fq.gz", sep = "")) %>%
  mutate(new_fastq = paste(Sample_ID, "_2.fq.gz", sep = ""))

R_2_rename_ready <- right_join(R_2, R_2_file_names, by = "fastq") %>%
  select(fastq, new_fastq)

write.table(R_2_rename_ready, file = "data/Sequencing/Raw_Fastqs/R_2_fastq_rename.txt", sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)

```

# Creating Sequencing File QC Meta # 
```{r}
# Import MultiQC Output Files 

multiqc_star <- read.delim("data/Sequencing/Porites_astreoides/MultiQC/multiqc_star.txt") %>%
  mutate(Sample = gsub("_RNASeq_2kenkel", '', Sample))


multiqc_star_alignment <- read.delim("data/Sequencing/Porites_astreoides/MultiQC/mqc_star_alignment_plot_1.txt") %>%
  mutate(Sample = gsub("_RNASeq_2kenkel", '', Sample))

Coral_ID_Names <- readxl::read_xlsx("data/Sequencing/Submission_and_Return_Sheets/Mote_Coral_ID_Names.xlsx")%>%
  select(Sample, Sample_ID = `Mote_Field/Frozen_SampleName`) 


# Combine MultiQC Output 

multiqc <- right_join(multiqc_star, multiqc_star_alignment, by = "Sample")

multiqc_Coral_Names <- right_join(Coral_ID_Names, multiqc, by = "Sample")


# Reformat Sequencing Submission Sheet 
Past_sub_data <- pre_seq %>%
  select(Sample_ID, RNATubeNumber = Sample, `Species Latin Name`, `Concentration(ng/ul)`, RIN, `Sample QC Results`, `Sample QC Memo`) %>%
  filter(`Species Latin Name` == "Porites astreoides")


# Merge Everything Else
y <- right_join(multiqc_Coral_Names, Past_sub_data, by = "Sample_ID") %>%
  select(Sample, Sample_ID, `Species Latin Name`: `Sample QC Memo`, total_reads:Unmapped..other)

dupes <- y %>%
  dplyr::count(Sample) %>%
  filter(n > 1)



```

